File: 05 - Formação React/01 - Programar com JS Orientado a Objetos/01- JS - Programando a Orientação a Objetos/ContaCorrente.js
import Cliente from "./Cliente.js";

class ContaCorrente{
    static numerosDeConta = 0;
    agencia;
    _cliente;

    // #saldo; Proposta de JS ainda não aprovada para atributos privados
    _saldo = 0; // Convenção da comunidade

    set cliente(novoValor){
        if(novoValor instanceof Cliente) this._cliente = novoValor;
    }

    get cliente(){ 
        return this._cliente;
    }

    get saldo(){
        return this._saldo;
    }

    constructor(agencia, cliente){
        this.agencia = agencia;
        this.cliente = cliente;
        ContaCorrente.numerosDeConta += 1;
    }

    sacar(valor){
        if(this._saldo >= valor) {
            this._saldo -= valor;
            return valor;
        }
    }

    depositar(valor){
        // Early Return
        if(valor <= 0) return;
        this._saldo += valor;
    }

    transferir(valor, conta){
        const valorSacado = this.sacar(valor);
        conta.depositar(valorSacado);
    }

}

export default ContaCorrente;

File: 14 - Formação C# e POO/02 - Orientação a Objetos/bytebank/ContaCorrente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace bytebank{
    public class ContaCorrente{
        public int numero_agencia;
        public string conta;
        public string titular;
        public double saldo;

        public ContaCorrente(int numero_agencia, string conta, string titular, double saldo) {
            this.numero_agencia = numero_agencia;
            this.conta = conta;
            this.titular = titular;
            this.saldo = saldo;
        }

        public void Depositar(double valor){
            this.saldo += valor;
        }

        public bool Sacar(double valor){
            if(valor <= this.saldo){
                this.saldo -= valor;
                return true;
            } else{
                return false;
            }
        }

        public bool Transferir(double valor, ContaCorrente destino){
            if(this.saldo < valor){
                return false;
            } else{
                this.Sacar(valor);
                destino.Depositar(valor);
                return true;
            }
        }


    }
}